source/MaterialXRenderGlsl/TextureBaker.cpp

```diff
@@ -0,0 +1,114 @@
+//
+// TM & (c) 2019 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
+// All rights reserved.  See LICENSE.txt for license.
+//
+
+#include <MaterialXRenderGlsl/TextureBaker.h>
+
+#include <MaterialXRenderGlsl/GLTextureHandler.h>
+#include <MaterialXRenderGlsl/GlslValidator.h>
+
+#include <MaterialXRender/LightHandler.h>
+#include <MaterialXRender/StbImageLoader.h>
+
+#include <MaterialXFormat/XmlIo.h>
+
+namespace MaterialX
+{
+
+void TextureBaker::bakeShaderInputs(ShaderRefPtr shaderRef, const FileSearchPath& searchPath, GenContext& context, const FilePath& outputFolder)
+{
+    if (!shaderRef)
+    {
+        return;
+    }
+
+    _searchPath = searchPath;
+    for (BindInputPtr bindInput : shaderRef->getBindInputs())
+    {
+        OutputPtr output = bindInput->getConnectedOutput();
+        if (output)
+        {
+            bakeGraphOutput(output, context, outputFolder);
+        }
+    }
+}
+
+void TextureBaker::bakeGraphOutput(OutputPtr output, GenContext& context, const FilePath& outputFolder)
+{
+    if (!output)
+    {
+        return;
+    }
+
+    _rasterizer = GlslValidator::create(_frameBufferRes);
+    StbImageLoaderPtr stbLoader = StbImageLoader::create();
+    GLTextureHandlerPtr imageHandler = GLTextureHandler::create(stbLoader);
+    imageHandler->setSearchPath(_searchPath);
+    _rasterizer->setImageHandler(imageHandler);
+    _rasterizer->initialize();
+
+    _generator = GlslShaderGenerator::create();
+
+    string outputName = createValidName(output->getNamePath());
+    ShaderPtr shader = _generator->generate(outputName + "_baker", output, context);
+
+    bool writeSrgb = output->getType() == "color3" || output->getType() == "color4";
+    _rasterizer->validateCreation(shader);
+    _rasterizer->renderScreenSpaceQuad(writeSrgb);
+
+    FilePath filename = outputFolder / FilePath(outputName + "_baked." + _fileSuffix);
+    _rasterizer->save(filename, false);
+}
+
+void TextureBaker::writeBakedDocument(ShaderRefPtr shaderRef, const FilePath& filename)
+{
+    if (!shaderRef)
+    {
+        return;
+    }
+
+    // Create document.
+    DocumentPtr bakedTextureDoc = createDocument();
+
+    // Create top-level elements.
+    NodeGraphPtr bakedNodeGraph = bakedTextureDoc->addNodeGraph("NG_baked");
+    MaterialPtr bakedMaterial = bakedTextureDoc->addMaterial("M_baked");
+    ShaderRefPtr bakedShaderRef = bakedMaterial->addShaderRef(shaderRef->getName() + "_baked", shaderRef->getAttribute("node"));
+    bakedNodeGraph->setColorSpace("srgb_texture");
```

Comment:
I'm not sure but as "srgb" output is an option shouldn't this also be set based on that option?

Reply:
I've updated the naming conventions and documentation here for clarity, but the encodeSrgb option only applies to the output of a single texture-space render.  This "srgb_texture" colorspace is being applied to the entire document, with color outputs being interpreted as sRGB, and scalar/vector outputs remaining linear.